numpy/core/src/multiarray/multiarraymodule.c

```diff
@@ -3795,36 +3795,6 @@ _PyArray_GetSigintBuf(void)
 #endif
 
 
-static PyObject *
-test_interrupt(PyObject *NPY_UNUSED(self), PyObject *args)
-{
-    int kind = 0;
-    int a = 0;
-
-    if (!PyArg_ParseTuple(args, "|i:test_interrupt", &kind)) {
-        return NULL;
-    }
-    if (kind) {
-        Py_BEGIN_ALLOW_THREADS;
-        while (a >= 0) {
-            if ((a % 1000 == 0) && PyOS_InterruptOccurred()) {
-                break;
-            }
-            a += 1;
-        }
-        Py_END_ALLOW_THREADS;
-    }
-    else {
-        NPY_SIGINT_ON
-        while(a >= 0) {
-            a += 1;
-        }
-        NPY_SIGINT_OFF
-    }
-    return PyInt_FromLong(a);
-}
```

Comment:
It might be worth extracting the `if` / `else` branches into the comment in `npy_interrupt`, showing how to replace uses of the `SIGINT` api.

Reply:
Ah, I thought that might require the GIL to be held, which seemed a bit useless for us. But, it seems like that has been fixed reading through all of: https://bugs.python.org/issue40826

Reply:
Didn't they end up deciding that it does require GIL?
https://github.com/python/cpython/pull/20618

Reply:
Oh, damn, so my first read was the right one, I guess I read the fact that it was backported, as that they made it not require the GIL (since I did not see why to backport otherwise) :(.

I think I will open a BPO, it would be nice if it was possible to at least be able to query this without the GIL if you pass in the threadstate or so...

Reply:
I opened https://bugs.python.org/issue41037 seems there already is a private function for this.  Not sure if we should/could petition more to make this function public, but I suppose it may be the best bet if we want interrupt capabilities at some point.

Reply:
Stumbled on it, so updated the comment now to mention `PyErr_CheckSignals()` and also link to the bpo.